trio_websocket/_impl.py

```diff
@@ -570,17 +636,33 @@ async def serve_websocket(handler, host, port, ssl_context, *,
         to finish the closing handshake before timing out.
     :param task_status: Part of Trio nursery start protocol.
     :returns: This function runs until cancelled.
-    '''
+    """
+    open_tcp_listeners: (
+        partial[Coroutine[Any, Any, list[trio.SocketListener]]]
+        | partial[Coroutine[Any, Any, list[trio.SSLListener[trio.SocketStream]]]]
+    )
     if ssl_context is None:
         open_tcp_listeners = partial(trio.open_tcp_listeners, port, host=host)
     else:
-        open_tcp_listeners = partial(trio.open_ssl_over_tcp_listeners, port,
-            ssl_context, host=host, https_compatible=True)
-    listeners = await open_tcp_listeners()
-    server = WebSocketServer(handler, listeners,
-        handler_nursery=handler_nursery, message_queue_size=message_queue_size,
-        max_message_size=max_message_size, connect_timeout=connect_timeout,
-        disconnect_timeout=disconnect_timeout)
+        open_tcp_listeners = partial(
+            trio.open_ssl_over_tcp_listeners,
+            port,
+            ssl_context,
+            host=host,
+            https_compatible=True,
+        )
+    listeners: list[trio.SSLListener[trio.SocketStream]] | list[trio.SocketListener] = (
+        await open_tcp_listeners()
+    )
```

Comment:
isn't the return type of `open_tcp_listeners()` inferred?

Reply:
It would be yes, but at the time I was making this it was having a lot of problems. I can remove the extra annotation if need be.

Reply:
Removed redundant annotation in b5233c8